colossalai/utils/activation_checkpoint.py

```diff
@@ -13,7 +13,8 @@ def copy_to_device(obj, device):
         # Notice:
         # When in no_grad context, requires_gard is False after movement
         ret = obj.to(device)
-        ret.requires_grad = obj.requires_grad
+        if ret.is_leaf or ret.retains_grad:
```

Comment:
I think activations to offload should be all non-leaf?

Reply:
yes... it is weird but we have met the problem.

Reply:
yes. I found that to() or cpu() or gpu() would change a leaf tensor to a non leaf one.